packages/flutter_tools/gradle/module_plugin_loader.gradle

```diff
@@ -20,15 +21,28 @@ if (pluginsFile.exists()) {
     object.plugins.android.each { androidPlugin ->
         assert androidPlugin.name instanceof String
         assert androidPlugin.path instanceof String
+        pluginsNames.add(androidPlugin.name)
         def pluginDirectory = new File(androidPlugin.path, 'android')
         assert pluginDirectory.exists()
         include ":${androidPlugin.name}"
         project(":${androidPlugin.name}").projectDir = pluginDirectory
```

Comment:
why not setting `buildDir` in the plugin project right here?

Reply:
I have tried to set buildDir here, but It's not work.
The plugin is added to host project by `include ":${androidPlugin.name}"`. 
and then the method: `org.gradle.initialization.DefaultProjectDescriptor#buildFile` in `gradle-core-6.1.1.jar` provide the buildDir location.
The build dir location provided by `org.gradle.initialization.DefaultProjectDescriptor#buildFile` is `getProjectDir + buildFileName` by default. 
And gradle only provide the setter for `buildFileName`. 
so I think It'll be difficult to modify the build dir by `org.gradle.initialization.DefaultProjectDescriptor`. the `projectsLoaded` code block will be a easier place.